src/sinks/util/vector_event.rs

```diff
@@ -1,42 +1,150 @@
 // Structs used for our vector event logs
 use regex::Regex;
 
+use std::collections::HashMap;
+use std::env;
+use once_cell::sync::OnceCell;
+use serde_json;
+
+use crate::event::Event;
+use crate::event::LogEvent;
+
+use crate::vector_lib::ByteSizeOf;
+
+pub static EVENT_LOG_METADATA_FIELD: OnceCell<String> = OnceCell::new();
+pub static EVENT_LOG_GRANULARITY_FIELDS: OnceCell<Vec<String>> = OnceCell::new();
+
+// Where we can find the log metadata object
+pub fn get_event_log_metadata_field() -> &'static String {
+    // Initialize the static variable once, or return the value if it's already initialized/computed
+    EVENT_LOG_METADATA_FIELD.get_or_init(|| {
+            env::var("EVENT_LOG_METADATA_FIELD").unwrap_or_else(|_| "".to_string())
+        }
+    )
+}
+
+// Within the log metadata object itself, these are fields we care to parse for
+pub fn get_event_log_granularity_fields() -> &'static Vec<String> {
+    // Initialize the static variable once, or return the value if it's already initialized/computed
+    EVENT_LOG_GRANULARITY_FIELDS.get_or_init(|| {
+            let vec_string: String = env::var("EVENT_LOG_GRANULARITY_FIELDS").unwrap_or_else(|_| "".to_string());
+            let keys: Vec<String> = serde_json::from_str(&vec_string).unwrap_or(Vec::new());
+            keys
+        }
+    )
+}
+
+/*
+* Struct to help track the count/size of events per unique combination of specified fields
+*/
+#[derive(Clone, Debug)]
+pub struct MetadataValuesCount {
+    pub value_map: HashMap<String, String>,
+    pub count: usize,
+    pub size: usize,
+}
+
 // Struct for vector send events (sending, uploaded)
 #[derive(Clone, Debug)]
-pub struct VectorSendEventMetadata {
+pub struct VectorEventLogSendMetadata {
     pub bytes: usize,
     pub events_len: usize,
     pub blob: String,
     pub container: String,
     pub topic: String,
+    // Count map here allows us to keep track of the count/size of events per combination of fields
+    // Key is a string encoding those combinations for ease of update
+    pub count_map: HashMap<String, MetadataValuesCount>,
 }
 
-impl VectorSendEventMetadata {
+impl VectorEventLogSendMetadata {
     pub fn emit_upload_event(&self) {
-        info!(
-            message = "Uploaded events.",
-            bytes = self.bytes,
-            events_len = self.events_len,
-            blob = self.blob,
-            container = self.container,
-            topic = self.topic,
-            // VECTOR_UPLOADED_MESSAGES_EVENT
-            vector_event_type = 4
-        );
+        self.emit_count_map("Uploaded events.", 4)
```

Comment:
Include a comment indicating what these event type values mean. Ditto for other `emit_count_map` calls below.

Reply:
Added